### PR TITLE
Add MeterProvider api

### PR DESCRIPTION
**Pull request by HudsonHumphries**
Implementation of the MeterProvider in the Metrics API. This class allows the user to set and get a meter provider, along with getting a meter from the MeterProvider.

The Provider class in Provider.h implements the functionality of the MeterProvider. This implementation is the same as the implementation for the TracerProvider already present in the repository. I also added another test on top of the tests from the TracerProvider

I stubbed out classes such as Meter, NoopMeter and NoopInstrument which made the tests correctly pass. Brandon’s PR will have these classes filled out, so I have removed the stubs from this PR.

**This PR will fail CI checks until [PR 162](https://github.com/open-telemetry/opentelemetry-cpp/pull/162) is merged. I had stubs in for classes and verified that it builds and tests pass. I will rebase once PR 162 is merged into the repo.**

### CODE REVIEW

File: api/test/metrics/meter_provider_test.cc [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+TEST(Provider, MultipleMeterProviders)
+{
+  auto tf = opentelemetry::nostd::shared_ptr<MeterProvider>(new TestProvider());
+  Provider::SetMeterProvider(tf);
+  auto tf2 = opentelemetry::nostd::shared_ptr<MeterProvider>(new TestProvider());
+  Provider::SetMeterProvider(tf2);
+
+  ASSERT_NE(tf, tf2);
+  ASSERT_NE(Provider::GetMeterProvider(), tf);
```
Comment (pyohannes) [inline api/test/metrics/meter_provider_test.cc:41]: This could probably just be a `ASSERT_EQ(tf2, Provider::GetMeterProvider())`?

Comment (HudsonHumphries) [reply api/test/metrics/meter_provider_test.cc:41]: I made this test to make sure that GetMeterProvider() changed from its previous MeterProvider. What you suggested is essentially the test above this one. If you think the MultipleMeterProviders test is unnecessary I can remove it

**Comment by HudsonHumphries** [comment]
I added meter stubs back into this PR and rebased with master

File: api/test/metrics/meter_provider_test.cc [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+TEST(Provider, MultipleMeterProviders)
+{
+  auto tf = opentelemetry::nostd::shared_ptr<MeterProvider>(new TestProvider());
+  Provider::SetMeterProvider(tf);
+  auto tf2 = opentelemetry::nostd::shared_ptr<MeterProvider>(new TestProvider());
+  Provider::SetMeterProvider(tf2);
+
+  ASSERT_NE(Provider::GetMeterProvider(), tf);
+}
```
Comment (reyang) [inline api/test/metrics/meter_provider_test.cc:41]: nit: add a blank line before EOF.

Comment (HudsonHumphries) [reply api/test/metrics/meter_provider_test.cc]: fixed

**Pull request merged**